### PR TITLE
lsp-protocol improvements

### PR DESCRIPTION
**Pull request by yyoncho**
1. Generated lsp-make-* contructors
2. Added few missing interfaces for jsonrpc itself
3. Added option to switch to plists

### CODE REVIEW

File: lsp-protocol.el [hunk 1]
```diff
@@ -31,6 +31,17 @@
...
+(defun lsp-keyword->symbol (keyword)
+  (intern (substring (symbol-name keyword) 1)))
+
+(defun lsp-keyword->string (keyword)
+  (substring (symbol-name keyword) 1))
+
+
+(defun lsp-use-plists ()
+  ;; change to t and recompile lsp-mode to force the usage of plists
+  nil)
```
Comment (ericdallo) [inline lsp-protocol.el:43]: Shouldn't this be `defvar` or `defcustom`?

File: lsp-protocol.el [hunk 2]
```diff
@@ -151,8 +205,26 @@ See `-let' for a description of the destructuring mechanism."
...
+               (JSONRequest (:params :method) nil)
+               (JSONError (:message :code) nil)
+               (ProgressParams (:token :value) nil)
+               (Edit (:kind) nil)
+               (WorkDoneProgress (:kind) nil)
+               (WorkDoneProgressBegin  (:kind :title) (:cancellable :message :percentage))
+               (WorkDoneProgressReport  (:kind) (:cancellable :message :percentage))
+               (WorkDoneProgressEnd  (:kind) (:message)))
+
+(defun dash-expand:&RangeToPoint (key source)
```
Comment (ericdallo) [inline lsp-protocol.el:225]: Nice!

File: lsp-protocol.el (hunk 1 above)
Comment (yyoncho) [reply lsp-protocol.el:43]: It can be, I changed it for some reason which I cannot remember right now.

File: lsp-protocol.el [hunk 3]
```diff
@@ -31,6 +31,17 @@
...
+
+(defun lsp-keyword->string (keyword)
+  (substring (symbol-name keyword) 1))
+
+
+(defun lsp-use-plists ()
+  ;; change to t and recompile lsp-mode to force the usage of plists
+  nil)
+
 (defmacro lsp-interface (&rest interfaces)
```
Comment (ericdallo) [inline lsp-protocol.el:45]: It does not need to be on this PR, but is it possible to make tests for this kind of macro?
This macro seems to be a little complex and a test could help us evolve in the future easily.

Comment (yyoncho) [reply lsp-protocol.el:45]: I started test suite for lsp-protocol.el

**Comment by yyoncho** [comment]
Closing the PR - we are going to work on top of plists branch.

**Pull request closed without merging**